src/inet/ZephyrSocketHacks.h

```diff
@@ -0,0 +1,59 @@
+#/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This header file defines BSD socket API functions which for various
+ *      reasons have not (yet) been implemented or exposed in Zephyr.
+ */
+
+#ifndef INET_ZEPHYR_SOCKET_HACKS_H
+#define INET_ZEPHYR_SOCKET_HACKS_H
```

Comment:
Please remove the word "hacks" from this PR.  If there is an issue with this implementation that needs to be addressed at some point, please file an issue.

Reply:
would "extensions" be acceptable?

Reply:
I think if you just dropped the word "HACKS_" from the symbol this would still work as intended, and would not characterize either side (CHIP or Zephyr) as needing a fix.

Reply:
Fixed in the current version